runtime/compiler/optimizer/TreeLowering.cpp

```diff
@@ -308,41 +308,44 @@ TR::TreeLowering::splitForFastpath(TR::Block* const block, TR::TreeTop* const sp
  * |BBEnd                         |         |
  * +------------------------------+         |
  * |BBStart (extension)           |         |
- * |ifacmpeq +->----------------------------+
- * |  iand                        |         |
- * |    iloadi ClassFlags         |         |
- * |      aloadi J9Class          |         |
- * |        aload rhs             |         |
- * |    iconst J9ClassIsValueType |         |
- * |  iconst 0                    |         |
- * |  GlRegDeps                   |         |
- * |    PassThrough x             |         |
- * |      ==> iconst 0            |         |
- * |    PassThrough ...           |         |
- * |BBEnd                         |         |
- * +------------------------------+         |
- * |BBStart (extension)           |         |
+ * |ifacmpeq +->-----------------------+    +
+ * |  iand                        |    |    |
+ * |    iloadi ClassFlags         |    |    |
+ * |      aloadi J9Class          |    |    |
+ * |        aload rhs             |    |    |
+ * |    iconst J9ClassIsValueType |    |    |
+ * |  iconst 0                    |    |    |
+ * |  GlRegDeps                   |    |    |
+ * |    PassThrough x             |    |    |
+ * |      ==> iconst 0            |    |    |
+ * |    PassThrough ...           |    |    |
+ * |BBEnd                         |    |    |
+ * +------------------------------+    |    |
```

Comment:
Should the `GlRegDeps` be added to the `BBEnd` of `checkRhsIsNotVT` block since  it reflects the register dependency to the merge block (`iRegLoad x`) ?